python/ray/includes/unique_ids.pxi

```diff
@@ -0,0 +1,986 @@
+"""This is a module for unique IDs in Ray.
+We define different types for different IDs for type safe.
+
+See https://github.com/ray-project/ray/issues/3721.
+"""
+
+from ray.includes.common cimport (
+    UniqueID as CUniqueID,
+    TaskID as CTaskID,
+    ObjectID as CObjectID,
+    JobID as CJobID,
+    FunctionID as CFunctionID,
+    ClassID as CClassID,
+    ActorID as CActorID,
+    ActorHandleID as CActorHandleID,
+    WorkerID as CWorkerID,
+    DriverID as CDriverID,
+    ConfigID as CConfigID,
+    ClientID as CClientID,
+)
+
+
+cdef class PyUniqueID:
+    cdef CUniqueID data
+    def __init__(self, object_id):
+        if object_id is None:
+            self.data = CUniqueID()
+        elif isinstance(object_id, bytes):
+            if len(object_id) != kUniqueIDSize:
+                raise ValueError("ID string needs to have length " + str(kUniqueIDSize))
+            self.data = CUniqueID.from_binary(object_id)
+        elif isinstance(object_id, PyUniqueID):
+            self.data = (<PyUniqueID>object_id).data
+        else:
+            raise TypeError("Unsupported type: " + str(type(object_id)))
+
+    @staticmethod
+    cdef from_native(const CUniqueID& cpp_id):
+        cdef PyUniqueID self = PyUniqueID.__new__(PyUniqueID)
+        self.data = cpp_id
+        return self
+
+    @staticmethod
+    def from_random():
```

Comment:
Let's remove this method everywhere. As mentioned in the previous object id refactor PR, it's not fork-safe, so we have to be very careful about using it from python...